pytorch_lightning/trainer/connectors/accelerator_connector.py

```diff
@@ -556,6 +560,23 @@ def _set_horovod_backend(self):
         else:
             self.num_processes = hvd.local_size()
 
+    def check_ipython_compatibility(self):
+        """
+        Raises a `MisconfigurationException` if the accelerator and/or plugin is not compatible with IPython
+        and code is run in an IPython kernel.
+        """
+        if self._distrib_type is None or self._distrib_type.is_ipython_compatible():
+            return
+        # check ipython env
+        if _module_available("IPython"):
+            from IPython import get_ipython
+            if get_ipython() is not None:
```

Comment:
noob question: when would this be None?

Reply:
`get_ipython()` will return `None` when the code is not run in a ipython environment(like notebook or ipython interactive shell)